src/ops/controlflow.rs

```diff
@@ -179,10 +179,8 @@ impl OpTrait for DataflowBlock {
         Some(EdgeKind::ControlFlow)
     }
 
-    fn dataflow_signature(&self) -> Option<FunctionType> {
-        Some(
-            FunctionType::new(type_row![], type_row![]).with_extension_delta(&self.extension_delta),
-        )
+    fn extension_delta(&self) -> ExtensionSet {
```

Comment:
The interesting thing here is that the default `extension_delta` did not delegate to `dataflow_signature` but just returned none, which AFAICS meant that extension inference was previously *ignoring* the extension delta for basic blocks: https://github.com/CQCL/hugr/blob/16dce1edb86374fbbe73fae648acdb030bd279c0/src/extension/infer.rs#L320